### ISSUE TITLE
Bazel: Unable to generate new app with bazel collection.

### ISSUE THREAD

**Issue by m0t0r** [angular/angular#28317]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/bazel

### Is this a regression?

 Not sure

### Description

With Angular 7.7.2, Bazel should now have support for SASS, so I decided to check it out and tried to generate a new app but the CLI never generates a new project.

## 🔬 Minimal Reproduction

 
Try to generate a new app with ```@angular/bazel``` collection.

```shell
ng new bazel-app --collection=@angular/bazel --defaults
```
or
```shell
ng new bazel-app --collection=@angular/bazel
```
Observe that CLI stops and never generates an app. 

## 🔥 Exception or Error
<pre><code>

 None

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 7.2.2
Node: 10.15.0
OS: darwin x64
Angular: 
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.12.2
@angular-devkit/core         7.2.2
@angular-devkit/schematics   7.2.2
@schematics/angular          7.2.2
@schematics/update           0.12.2
rxjs                         6.3.3
typescript                   3.2.2

</code></pre>

```shell
➜  ~ npm view @angular/bazel version
7.2.2
```
**Anything else relevant?**

Friendly pinging @mgechev :)

**Comment by kyliau** [comment]
@m0t0r is there an existing app in the directory where you invoked `ng new`?
My speculation is that CLI takes a long time to generate an internal tree that represents the directory structure when there is `node_modules` in the directory where `ng new` is invoked.
Could you please try this on an empty directory and see if the problem goes away?
If not, please also try installing the `@next` version of `@angular/cli`.

**Comment by m0t0r** [comment]
@kyliau you are absolutely right, I was trying to generate the app in my projects directory with lots of apps which have ```node_modules```. When I created an empty directory and tried to generate new app with bazel collection, it worked very quickly. But when I tried to run the app with ```ng serve```I got errors:

```shell
[3/5] Fetching packages...
info @bazel/buildifier-linux_x64@0.19.2: The platform "darwin" is incompatible with this module.
info "@bazel/buildifier-linux_x64@0.19.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
[5/5] Building fresh packages...
error /private/var/tmp/_bazel_m0t0r/2d1b54022445b9684fa5af412dcccbde/external/ngdeps/node_modules/@bazel/typescript: Command failed.
Exit code: 1
Command: node ./check_version.js
Arguments: 
Directory: /private/var/tmp/_bazel_m0t0r/2d1b54022445b9684fa5af412dcccbde/external/ngdeps/node_modules/@bazel/typescript
Output:
/private/var/tmp/_bazel_m0t0r/2d1b54022445b9684fa5af412dcccbde/external/ngdeps/node_modules/@bazel/typescript/check_version.js:73
    throw new Error(`Expected build_bazel_rules_typescript to be version ${
    ^

Error: Expected build_bazel_rules_typescript to be version 0.22.0
    at Object.<anonymous> (/private/var/tmp/_bazel_m0t0r/2d1b54022445b9684fa5af412dcccbde/external/ngdeps/node_modules/@bazel/typescript/check_version.js:73:11)
```

I got same error with ```@angular/cli@next```.

**Comment by mgechev** [comment]
@m0t0r this looks like a mismatch of the versions of `rules_typescript` in the workspace file and `package.json`. Would you verify that both are 0.22.1 or 0.22.0?

**Comment by m0t0r** [comment]
@mgechev , @kyliau 

So the issue seems to be in:
```/private/var/tmp/_bazel_m0t0r/2d1b54022445b9684fa5af412dcccbde/external/ngdeps/node_modules/@bazel/typescript``` the version value in ```package.json``` is "0.22.0", and the check by ```@bazel/typescript/check_version.js```fails because of that. I didn't have enough time to look deeper to figure out what still makes it to install version "0.22.0" over "0.22.1". Note, inside the workspace file and ```package.json```of the the CLI the version is set correctly to "0.22.1".

When I tried to set version to "0.22.0" in both the workspace file and ```package.json```of the the CLI I got following errors:

```shell
INFO: Invocation ID: db27ddd9-4e00-48b0-adb2-7b30102387bc
ERROR: /Users/m0t0r/WebstormProjects/bzl/bazel-app/src/BUILD.bazel:67:1: //src:devserver: no such attribute 'index_html' in 'ts_devserver' rule
ERROR: error loading package 'src': Package 'src' contains errors
INFO: Elapsed time: 0.077s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
```

**Comment by Felx-B** [comment]
I had the same issue, resolved by upgrading angular version to 7.2.2 in WORKSPACE file

**Comment by kylecordes** [comment]
I also hit the problem with " I was trying to generate the app in my projects directory with lots of apps which have node_modules." - A long freeze and then finally failing with an error. Although easy to work around, this is also a reasonably surprising bug, my mental model of "ng new" would be that it creates the new directory and does all the work there, I don't know why it would even have the opportunity or capability of looking around other directories than the one I told it to create!

**Comment by kyliau** [comment]
https://github.com/angular/angular/pull/29148 deleted `bazel-workspace` schematics. It scans the filesystem for certain files, which is the primary cause of the slow operation. 
It will be released in the next beta release, v8.0.0-beta.8.

**Comment by kyliau** [comment]
Closing this since v8-beta.8 was released.
As a reminder, if you're trying out Bazel, please make sure to always install `@angular/cli@next` and `@angular/bazel@next`.
